### ISSUE TITLE
twoMonthOverlap

### ISSUE THREAD

**Issue by linyao1999** [brianpm/wavenumber_frequency#2]
Hi Brian,

Thanks for the sharing of the codes. I wonder if you could double-check the parameter 'twoMonthOverlap' or 'noverlap' in your function called spacetime_power(). You roll the data using the parameter in line 381, which is the stride instead of the overlap length. 

Best,
Lin

**Comment by brianpm** [comment]
Thanks for the comment. I'll have to take a closer look. Do you think that it should be:

```
x_win = x_roll.construct("segments", stride=(segsize-noverlap)).dropna("time")
```
?

**Comment by linyao1999** [comment]
Yes, that is what I am thinking. Besides, noverlap should be multiplied by spd as well.

**Comment by brianpm** [comment]
Addressed by d707d473a83bf36d1ea214d3bca463814dbab00d

This turned out to be slightly more subtle than we thought at first. The `dropna("time")` bit was actually obscuring an indexing inconsistency that was sneaking in. I believe that the correct fix is to replace that section with this:
```python
x_roll = data.rolling(time=segsize, min_periods=segsize)  # WK99 use 96-day window
assert segsize-noverlap > 0, f"Error, inconsistent specification of segsize and noverlap results in stride of {segsize-noverlap}, but must be > 0."
x_win = x_roll.construct("segments")
x_win = x_win.isel(time=slice(segsize-1,None,segsize-noverlap))  
```
This comes at the cost of `construct`ing a bigger DataArray first and then slicing it. The slicing in the last line, however, is the correct way to start at the beginning of the time series and stride to get the correct overlap.
